pkg/instrumentation/router.go

```diff
@@ -25,16 +25,21 @@ type Tracer struct {
 	Enabled     bool
 	Environment string
 	Options     []tracer.StartOption
+
+	globalServiceName string
 }
 
 // NewTracer returns a new tracer with the giver configuration and an optional
 // list of ddtrace's tracer.StartOptions.
 //
 // NewTracer assigns universal the version of the service that is running, and will be applied to all spans,
 // regardless of whether span service name and config service name match.
-func NewTracer(config *Config, options ...tracer.StartOption) *Tracer {
+func NewTracer(appName string, config *Config, options ...tracer.StartOption) *Tracer {
+	serviceName := globalServiceName(appName)
```

Comment:
[q]: what do you think if instead of introducing a new argument to the function (which will force us to update every other downstream call), add a new field to config with the `ServiceName` name and use that instead?

Reply:
Tracer service name here was introduced by combining `applicationName` and `datadogPostfix` (which make sense as long as we require a federated metrics by grouped instances), following the order, applicationName was needed in `NewTracer` and as long as application name is not relevant to tracer config, this came out as design.
However, Having `service name` in tracer config allows us more flexibility. I will move changes to related config.

PS: I believe we agreed upon having braking changes cannot be the reason to reject a change.